### PR TITLE
Added table page size preference

### PR DESCRIPTION
**Pull request by mklanjsek**
- Added a new preference to set table page size with default value of 10,
- Added Reset button so the preferences can be restored to default state,
- Rearranged the panels - moved the navigation preferences to separate panel, 
- Refactored preferences handling to make it easier to add new and maintain existing 

Luis and Milan

### CODE REVIEW

File: web/src/app/modules/shared/components/presentation/preferences/preferences.component.html [hunk 1]
```diff
@@ -60,6 +60,11 @@ <h4>{{ section.name }}</h4>
                         </div>
                       </div>
                     </ng-container>
+                    <ng-container *ngSwitchCase="'dropdown'">
+                      <label>{{ element.label }}</label>
+                      <app-view-dropdown [view]="element" (selectedValue)="onDropDownValueChange($event, element.name)">
```
Comment (GuessWhoSamFoo) [inline web/src/app/modules/shared/components/presentation/preferences/preferences.component.html:65]: Why is dropdown used here over select (https://clarity.design/angular-components/select/)?

Comment (lenriquez) [reply web/src/app/modules/shared/components/presentation/preferences/preferences.component.html:65]: This is a good question, we are not using clarity here, I saw this component and just used

Comment (mklanjsek) [reply web/src/app/modules/shared/components/presentation/preferences/preferences.component.html:65]: I am OK with using our dropdown here, there a lots of benefits of dogfooding.

Comment (GuessWhoSamFoo) [reply web/src/app/modules/shared/components/presentation/preferences/preferences.component.html:65]: Implicitly this means using dropdown over select for all of preferences in order to maintain a consistent design. Are we okay with that?

Comment (wwitzel3) [reply web/src/app/modules/shared/components/presentation/preferences/preferences.component.html:65]: I think we should be consistent within the Clarity guidelines.

https://clarity.design/angular-components/dropdown/#dropdown-vs-select

**Comment by mklanjsek** [comment]
Clarity selector looks good to me, thanks @lenriquez!

@GuessWhoSamFoo @wwitzel3 PTAL

**Review by GuessWhoSamFoo — APPROVED** [review]
lgtm

**Pull request merged**